pkg/gofr/datasource/pubsub/mqtt/mqtt.go

```diff
@@ -59,10 +59,8 @@ type subscription struct {
 // New establishes a connection to MQTT Broker using the configs and return pubsub.MqttPublisherSubscriber
 // with more MQTT focused functionalities related to subscribing(push), unsubscribing and disconnecting from broker.
 func New(config *Config, logger Logger, metrics Metrics) *MQTT {
-	var options *mqtt.ClientOptions
```

Comment:
@Umang01-hash This one breaks a test

Reply:
Oups, it might be because of the changes I suggested

🤔😱

Reply:
@ccoVeille there's problem with getDefaultClientOpts that's mutable. Feel free to improve on that

Reply:
I with there was a linter like "Don't mutate anything in method caller getSomething(*Obj)"

Reply:
Indeed